nsxt/data_source_nsxt_ip_pool.go

```diff
@@ -0,0 +1,87 @@
+/* Copyright © 2017 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	api "github.com/vmware/go-vmware-nsxt"
+	"github.com/vmware/go-vmware-nsxt/manager"
+	"net/http"
+)
+
+func dataSourceNsxtIPPool() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceNsxtIPPoolRead,
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:        schema.TypeString,
+				Description: "Unique ID of this resource",
+				Optional:    true,
+				Computed:    true,
+			},
+			"display_name": {
+				Type:        schema.TypeString,
+				Description: "The display name of this resource",
+				Optional:    true,
+				Computed:    true,
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Description: "Description of this resource",
+				Optional:    true,
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func dataSourceNsxtIPPoolRead(d *schema.ResourceData, m interface{}) error {
+	// Read IP Pool by name or id
+	nsxClient := m.(*api.APIClient)
+	objID := d.Get("id").(string)
+	objName := d.Get("display_name").(string)
+	var obj manager.IpPool
+	if objID != "" {
+		// Get by id
+		objGet, resp, err := nsxClient.PoolManagementApi.ReadIpPool(nsxClient.Context, objID)
+
+		if err != nil {
```

Comment:
order of these error checks was recently reversed in https://github.com/terraform-providers/terraform-provider-nsxt/pull/191